torchvision/models/googlenet.py

```diff
@@ -60,37 +60,115 @@ def googlenet(pretrained=False, progress=True, **kwargs):
     return GoogLeNet(**kwargs)
 
 
+class BasicConv2d(nn.Module):
+
+    def __init__(self, in_channels, out_channels, **kwargs):
+        super(BasicConv2d, self).__init__()
+        self.conv = nn.Conv2d(in_channels, out_channels, bias=False, **kwargs)
+        self.bn = nn.BatchNorm2d(out_channels, eps=0.001)
+
+    def forward(self, x):
+        x = self.conv(x)
+        x = self.bn(x)
+        return F.relu(x, inplace=True)
+
+
+class Inception(nn.Module):
+
+    def __init__(self, in_channels, ch1x1, ch3x3red, ch3x3, ch5x5red, ch5x5, pool_proj,
+            basic_conv2d=BasicConv2d):
+        super(Inception, self).__init__()
+
+        self.branch1 = basic_conv2d(in_channels, ch1x1, kernel_size=1)
+
+        self.branch2 = nn.Sequential(
+            basic_conv2d(in_channels, ch3x3red, kernel_size=1),
+            basic_conv2d(ch3x3red, ch3x3, kernel_size=3, padding=1)
+        )
+
+        self.branch3 = nn.Sequential(
+            basic_conv2d(in_channels, ch5x5red, kernel_size=1),
+            basic_conv2d(ch5x5red, ch5x5, kernel_size=3, padding=1)
+        )
+
+        self.branch4 = nn.Sequential(
+            nn.MaxPool2d(kernel_size=3, stride=1, padding=1, ceil_mode=True),
+            basic_conv2d(in_channels, pool_proj, kernel_size=1)
+        )
+
+    def _forward(self, x):
+        branch1 = self.branch1(x)
+        branch2 = self.branch2(x)
+        branch3 = self.branch3(x)
+        branch4 = self.branch4(x)
+
+        outputs = [branch1, branch2, branch3, branch4]
+        return outputs
+
+    def forward(self, x):
+        outputs = self._forward(x)
+        return torch.cat(outputs, 1)
+
+
+class InceptionAux(nn.Module):
```

Comment:
Same comment here, is this code movement necessary? If not, can you revert this? This makes reviewing harder

Reply:
It's necessary as the class needs to be defined before using it in GoogLeNet class. Do you know if there's any way to work around this to avoid code movement?